server/src/main/java/org/elasticsearch/action/admin/indices/recovery/RecoveryResponse.java

```diff
@@ -87,6 +86,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 if (recoveryStates == null || recoveryStates.size() == 0) {
                     continue;
                 }
+                builder.startObject();
```

Comment:
Doesn't this result in an empty builder when `!hasRecoveries()`? Shouldn't we be returning an empty object instead?
( Maybe it is ok, I don't what the convention is.. )

Reply:
@olcbean Yes, you are right. The `builder.startObject();` `builder.endObject();` pair should be put outside the condition I think. Just like what it was previously.